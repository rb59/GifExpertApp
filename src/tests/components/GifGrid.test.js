import GifGrid from "../../components/GifGrid";
import {shallow} from 'enzyme';
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {

    
    const category = 'One punch';
    

    test('debe renderizar correctamente', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
        
    });
    
    test('debe mostrar items al cargar imagenes', () => {

        const gifs = [{
            id: '123',
            url: 'http://prueba.gif',
            title: 'prueba',
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
        
    });
    
    
})
