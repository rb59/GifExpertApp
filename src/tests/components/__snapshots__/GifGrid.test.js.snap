// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe mostrar items al cargar imagenes 1`] = `
<Fragment>
  <h3>
    One punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="123"
      key="123"
      title="prueba"
      url="http://prueba.gif"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe renderizar correctamente 1`] = `
<Fragment>
  <h3>
    One punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
